bot/cogs/crashes.py

```diff
@@ -1,55 +1,29 @@
 from fred_core_imports import *
 from libraries import common, createembed
 
-import errno
-from functools import wraps
 import nextcord.ext.commands as commands
-from PIL import Image, ImageEnhance
+from PIL import Image, ImageEnhance, UnidentifiedImageError
 from pytesseract import image_to_string, TesseractError
 import zipfile
-from urllib.request import urlopen
 from concurrent.futures import ThreadPoolExecutor
 from time import strptime
+from typing import AsyncIterator
 
 
-
-def timeout(seconds=2, error_message=os.strerror(errno.ETIME)):
-    def decorator(func):
-        def _handle_timeout(signum, frame):
-            raise TimeoutError(error_message)
-
-        def wrapper(*args, **kwargs):
-            signal.signal(signal.SIGALRM, _handle_timeout)
-            signal.setitimer(signal.ITIMER_REAL, seconds)  # used timer instead of alarm
-            try:
-                result = func(*args, **kwargs)
-            finally:
-                signal.alarm(0)
-            return result
-
-        return wraps(func)(wrapper)
-
-    return decorator
-
-
-@timeout(5)
-def timedregexsearch(*args, **kwargs):
+async def regex_with_timeout(*args, **kwargs):
     try:
-        return re.search(*args, **kwargs)
```

Comment:
Isn't there some alternative to `to_thread` here? Creating threads and changing context is relatively heavy